product_class/product.rb

```diff
@@ -0,0 +1,101 @@
+class Product
+  VALID_ATR = %i(name value brand description quantity)
+
+  attr_reader :all
+
+  def initialize
+    @all = []
+  end
+
+  def create(id:, name:, value:, brand:, **optional_info)
+    info = {
+      id: id,
+      name: name,
+      value: value,
+      brand: brand
+    }
+    if optional_info[:description]
+      info.merge!(description: optional_info[:description])
+    end
+    if optional_info[:quantity]
+      info.merge!(quantity: optional_info[:quantity])
+    end
+
+    @all << info
+  end
+
+  def count
+    @all.size
+  end
+
+  def find(id:)
+    @all.find { |product| id == product[:id] }
+  end
+
+  def where(**filters)
+    products = []
+
+    @all.each do |product|
+      check = true
+      filters.each do |atr, value|
+        check = false unless product[atr] == value
+      end
+      products << product if check
+    end
+
+    products
+  end
+
+  def update(id:, **info_update)
+    @all.find do |product|
```

Comment:
I meant something like this. The update part is not responsability of the block in the find method.
```suggestion
    product = find(id: id)
    
    return unless product
    
    info_update.each do |atr, value_update|
      product[atr] = value_update if VALID_ATR.include?(atr)
    end
```